### DISCUSSION TITLE
How to pass state value to getServerSideProps?

### DISCUSSION THREAD

**Discussion by YouyiRhode** [vercel/next.js#27774] [Help]
I put the input value into the component state , and I want to pass that value to getServerSideProps to fetch data. How could I pass state value to getServerSideProps?

**Comment by maful** [comment]
`getServerSideProps` is runs at the request time, if you pass the value into a component, I think you should consider using `useEffect` to fetch the data in that component

https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering

**Reply by YouyiRhode** [reply]
My app is user input actor name and then send the name to getServerSideProps function and fetch data from the movie database. Should be ok use getServerSideProps to fetch data right?
